components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1541,6 +1557,45 @@ private String handleOAuthAuthorizationRequest(OAuthMessage oAuthMessage)
         }
     }
 
+    /**
+     * Returns an instance of OAuthAuthzRequest. If the configured classname is invalid the default implementation
+     * will be returned.
+     *
+     * @param oauthAuthzRequestClassName configured OAuthAuthzRequest class name.
+     * @param request http servlet request.
+     * @return instance of OAuthAuthzRequest.
+     * @throws OAuthProblemException thrown when initializing the OAuthAuthzRequestClass instance.
+     * @throws OAuthSystemException thrown when initializing the OAuthAuthzRequestClass instance.
+     */
+    private OAuthAuthzRequest getOAuthAuthzRequest(String oauthAuthzRequestClassName, HttpServletRequest request)
+            throws OAuthProblemException, OAuthSystemException {
+
+        OAuthAuthzRequest oAuthAuthzRequest;
+        try {
+            // Validations will be performed when initializing the class instance.
+            Class clazz = Thread.currentThread().getContextClassLoader().loadClass(oauthAuthzRequestClassName);
```

Comment:
better to pass the Class itself as the param to avoid the loading at each request,

Also you may be able to restrict the class to be of this type by defining the param type as,
Class<? extends OAuthAuthzRequest>

Reply:
Fixed with 79bab73a362a2021f071c540d7aed7a385ff3817